comms/src/lib.rs

```diff
@@ -17,6 +17,7 @@
 #![recursion_limit = "512"]
 // Allow `type Future = impl Future`
 #![feature(type_alias_impl_trait)]
+#![feature(min_type_alias_impl_trait)]
```

Comment:
This feature [may not be stabilised for a while](https://github.com/rust-lang/rust/issues/63063#issuecomment-795682305) so you might be hampering the effort to move Tari to stable Rust.

Reply:
It is the same issue the feature above which is critical to large portions of the code, they share the same tracking Issue in the Rust codebase.